lib/matplotlib/contour.py

```diff
@@ -1213,10 +1211,17 @@ def _process_levels(self):
         # (Colorbar needs this even for line contours.)
         self._levels = list(self.levels)
 
+        tempLocator = Locator()
+        if self.logscale:
+            tempLocator = LogLocator()
+        if self.locator is not None:
+            tempLocator = self.locator
```

Comment:
Actually, I thought it over, you are right. LogNorm is probably kind of corner case whereas other existing norms are more well-behaved so maybe doing it your way is better. I will fix this up.